basyx/aas/model/base.py

```diff
@@ -610,12 +610,17 @@ def __init__(self):
     def __repr__(self) -> str:
         reversed_path = []
         item = self  # type: Any
+        from .submodel import SubmodelElementList
         while item is not None:
             if isinstance(item, Identifiable):
-                reversed_path.append(str(item.id))
+                reversed_path.append(item.id)
                 break
             elif isinstance(item, Referable):
-                reversed_path.append(item.id_short)
+                if isinstance(item.parent, SubmodelElementList):
+                    reversed_path.append(f"{item.parent.id_short}[{item.parent.value.index(item)}]")
+                    item = item.parent
```

Comment:
Wouldn't this lead to having the `SubmodelElementList.id_short` twice in the `__repr__`?

Let's say we have an `example_sel` with a child on index one, on which we call `__repr__`.
Then we would add in line 620 `example_sel[1]` to the reversed path, before setting the `item` to `example_sel` and adding the `example_sel` to the reversed path again?

Reply:
Nope, because we do `item = item.parent` again on line 624. Yes, it's a bit confusing, we could also change this to `item = item.parent.parent` and move the `item = item.parent` on line 624 into the `else` block.

Reply:
Ah, I see. I guess for now it's fine to leave it this way.